BraintreeCore/src/test/java/com/braintreepayments/api/PaymentMethodClientUnitTest.java

```diff
@@ -0,0 +1,143 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.isNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@RunWith(RobolectricTestRunner.class)
+public class PaymentMethodClientUnitTest {
+
+    @Captor
+    ArgumentCaptor<List<BraintreeNonce>> paymentMethodNoncesCaptor;
+
+    @Before
+    public void beforeEach() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void getPaymentMethodNonces_whenDefaultFirstIsFalse_sendsPaymentMethodsGETRequest() {
+        BraintreeClient braintreeClient = new MockBraintreeClientBuilder()
+                .sessionId("sample-session-id")
+                .build();
+
+        PaymentMethodClient sut = new PaymentMethodClient(braintreeClient);
+        sut.getPaymentMethodNonces(false, mock(GetPaymentMethodNoncesCallback.class));
+
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+        verify(braintreeClient).sendGET(captor.capture(), any(HttpResponseCallback.class));
+
+        String url = captor.getValue();
+        assertEquals("/v1/payment_methods?default_first=false&session_id=sample-session-id", url);
+    }
+
+    @Test
+    public void getPaymentMethodNonces_whenDefaultFirstIsTrue_sendsPaymentMethodsGETRequest() {
+        BraintreeClient braintreeClient = new MockBraintreeClientBuilder()
+                .sessionId("sample-session-id")
+                .build();
+
+        PaymentMethodClient sut = new PaymentMethodClient(braintreeClient);
+        sut.getPaymentMethodNonces(true, mock(GetPaymentMethodNoncesCallback.class));
+
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+        verify(braintreeClient).sendGET(captor.capture(), any(HttpResponseCallback.class));
+
+        String url = captor.getValue();
+        assertEquals("/v1/payment_methods?default_first=true&session_id=sample-session-id", url);
+    }
+
+    @Test
+    public void getPaymentMethodNonces_parsesCards() {
+        BraintreeClient braintreeClient = new MockBraintreeClientBuilder()
+                .sessionId("sample-session-id")
+                .sendGETSuccessfulResponse(Fixtures.PAYMENT_METHODS_GET_PAYMENT_METHODS_RESPONSE)
+                .build();
+
+        PaymentMethodClient sut = new PaymentMethodClient(braintreeClient);
+
+        GetPaymentMethodNoncesCallback callback = mock(GetPaymentMethodNoncesCallback.class);
+        sut.getPaymentMethodNonces(false, callback);
+
+        verify(callback).onResult(paymentMethodNoncesCaptor.capture(), (Exception) isNull());
+
+        List<BraintreeNonce> paymentMethodNonces = paymentMethodNoncesCaptor.getValue();
+        assertEquals("Visa", paymentMethodNonces.get(0).getTypeLabel());
+        assertEquals("ending in ••11", paymentMethodNonces.get(0).getDescription());
```

Comment:
So, we had [an issue in iOS Drop-in](https://github.com/braintree/braintree-ios-drop-in/issues/188) about the `description` on payment method nonces not being localized. For cards specifically, the description that comes back from the Gateway is in English (i.e., "ending in ••11").

On iOS, we decided to remove `description` from the payment method nonce class in the base SDK, and add a description property to an extension on payment method nonces in Drop-in. https://github.com/braintree/braintree-ios-drop-in/blob/76a657e1b95f78643dfd4395f8e2c79d1c08c3b1/Sources/BraintreeDropIn/BTPaymentMethodNonce%2BDropIn.m#L23

We have some client-side logic to check the type of the nonce and use the relevant property for that type. (So for Venmo, it's the username, for PayPal it's the email address, for cards it's just the last four digits.)

I'm thinking we should do something similar in Android?

Reply:
Yeah I agree we definitely should do something similar to match iOS localization. That'll be super useful for custom integrations too if we handle it in the base of the sdk.

Reply:
This can probably be completed in a separate PR, yeh?

Reply:
We've added a TODO for this 👍